### PR TITLE
DOP-1151: prep Compass for building/publishing with Snooty

### PR DESCRIPTION
**Pull request by schmalliso**
## Changes

- Ports substitutions fron conf.py to snooty.toml (I only brought over the ones you were actually using to avoid excessive cruft)
- Added relevant pages to the toc_landing_pages so they were openable from the TOC
- Other necessary structural updates to snooty.toml and worker.sh
- Fixes a broken figure embed, and swaps a legacy `manual-next` for `manual` since wildcard indexes are very much established in MongoDB 😁 
- Accidentally linted the steps-create-index.yaml file. Sorry about that.

Staged: https://docs-mongodbcom-staging.corp.mongodb.com/compass/docsworker-xlarge/snooty/
Workerpool log: https://workerpool-boxgs.mongodbstitch.com/pages/job.html?jobId=5f3b10ae2f9a5cfb5bce547e

## Warnings to Address

```
WARNING(query-plan.txt:20ish): "figure" requires the following options: alt
```
All figures and images should have alt text descriptions. We're now complaining when images don't. For a primer on how to write a good alt tag, refer to https://webaim.org/techniques/alttext/ and/or https://www.w3.org/WAI/tips/writing/#write-meaningful-text-alternatives-for-images

```
WARNING(includes/styles/corrections.rst:0ish): Directive "raw" has been deprecated
```
Raw has been deprecated! I've no idea what this file is supposed to do, if we have styling corrections that are needed, please file a ticket so we can address them via the actual CSS rather than this cheeky workaround.

## Warnings to Not Worry About

```
WARNING(install.txt:12ish): Directive "tabs" has been deprecated
```
Tabs haven't been deprecated - there's [a ticket](https://jira.mongodb.org/browse/DOP-1299) to make this message more usefully clear. The legacy YAML style tab syntax *has* been deprecated, though, as has the pillstrip thing.

```
warn There are conflicting field types in your data.

If you have explicitly defined a type for those fields, you can safely ignore this warning message.
Otherwise, Gatsby will omit those fields from the GraphQL schema.

If you know all field types in advance, the best strategy is to explicitly define them with the
`createTypes` action, and skip inference with the `@dontInfer` directive.
SitePage.context.metadata.toctree.children.children.children.title:
 - type: array
   value: [ ..., { type: [32m'text'[39m, position: [36m[Object][39m, value: [32m'View Documents'[39m }, ... ]
 - type: string
   value: [32m'Sort Returned Documents'[39m
```

There's something slightly wonky in how we're handling TOCs on the back end -- this isn't actually causing anything wonky in the rendered site, but is producing the following warning:

Don't worry about it ([DOP-981](https://jira.mongodb.org/browse/DOP-981) will be dealing with it).

### CODE REVIEW

File: snooty.toml [hunk 1]
```diff
@@ -1,6 +1,21 @@
...
+intersphinx = ["https://docs.mongodb.com/manual/objects.inv"]
+
+toc_landing_pages = ["/install", "/connect", "/instance", "/query/filter", "/query/queries",
+                     "/documents", "/schema", "/aggregation-pipeline-builder"]
 
 [constants]
-version = "1.18"
-current-version = "1.17.0"
 download-page = "`downloads page <https://www.mongodb.com/download-center/compass?tck=docs_compass>`__"
+current-version = "1.20.4"
```
Comment (jeff-allen-mongo) [inline snooty.toml:11]: [nit] Does this affect what get shows up when you hover over the tab in your browser? We typically don't document specific minor versions, so maybe this should be "1.21" ?

Comment (schmalliso) [reply snooty.toml:11]: I used what you have for the {{current_version}} source constant in conf.py currently. It appears that source constant is used in install.txt for the RHEL install (line 202) YUM command.

Comment (jeff-allen-mongo) [reply snooty.toml:11]: d'oh I need to update that 😅

**Comment by schmalliso** [comment]
Pleased to report that the pipeline issue has indeed been resolved. Once we have icons handled, we'll be all clear! Yay!

**Comment by jeff-allen-mongo** [comment]
Exciting! 🚀

**Pull request merged**